workspace/__ardulib__/Graphics/Canvas.inl

```diff
@@ -0,0 +1,36 @@
+
+void Canvas::setPosition(int16_t x, int16_t y, int16_t w, int16_t h) {
+    _a = XVector2<int16_t>(x, y);
+    _w = w;
+    _h = h;
+}
+
+void Canvas::setStyle(XColor bgColor, XColor fgColor) {
+    _bgColor = bgColor;
+    _fgColor = fgColor;
+}
+
+const XColor* Canvas::getFGColor() const {
+    return &_fgColor;
+}
+
+XVector2<int16_t> Canvas::getCanvasSetPoint() const {
+    return _a;
+}
+
+int16_t Canvas::getCanvasWidth() const {
+    return _w;
+
+}
+int16_t Canvas::getCanvasHeight() const {
+    return _h;
+}
+
+void Canvas::renderScanline(XRenderer* renderer, int16_t scanline, int16_t xmin, int16_t xmax, uint16_t* buffer) {
+
+    uint16_t color = xcolorTo565(_bgColor);
+
+    for (int16_t x = 0; x < _w; x++)
+        buffer[x] = color;
```

Comment:
This should take `xmin` and `xmax` into account for partial updates (for whatever reason).